library/ssl_tls13_server.c

```diff
@@ -97,6 +98,319 @@ static int ssl_tls13_parse_key_exchange_modes_ext( mbedtls_ssl_context *ssl,
     ssl->handshake->tls13_kex_modes = ke_modes;
     return( 0 );
 }
+
+#define SSL_TLS1_3_OFFERED_PSK_NOT_MATCH   0
+#define SSL_TLS1_3_OFFERED_PSK_MATCH       1
+static int ssl_tls13_offered_psks_check_identity_match(
+               mbedtls_ssl_context *ssl,
+               const unsigned char *identity,
+               uint16_t identity_len )
+{
+    /* Check identity with external configured function */
+    if( ssl->conf->f_psk != NULL )
+    {
+        if( ssl->conf->f_psk(
+                ssl->conf->p_psk, ssl, identity, identity_len ) == 0 )
+        {
+            return( SSL_TLS1_3_OFFERED_PSK_MATCH );
+        }
+        return( SSL_TLS1_3_OFFERED_PSK_NOT_MATCH );
+    }
+
+    /* Check identity with pre-configured psk */
+    if( identity_len == ssl->conf->psk_identity_len &&
+        mbedtls_ct_memcmp( ssl->conf->psk_identity,
+                            identity, identity_len ) == 0 )
+    {
+        mbedtls_ssl_set_hs_psk( ssl, ssl->conf->psk, ssl->conf->psk_len );
+        return( SSL_TLS1_3_OFFERED_PSK_MATCH );
+    }
+
+    /* Add session ticket here */
+
+    return( SSL_TLS1_3_OFFERED_PSK_NOT_MATCH );
+}
+
+static int ssl_tls13_offered_psks_check_binder_match(
+               mbedtls_ssl_context *ssl,
+               const unsigned char *binder,
+               uint16_t binder_len,
+               const unsigned char *psk,
+               size_t psk_len )
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    int psk_type;
+    mbedtls_md_type_t md_alg =
+        binder_len == 32 ? MBEDTLS_MD_SHA256 : MBEDTLS_MD_SHA384 ;
+    psa_algorithm_t psa_md_alg = mbedtls_psa_translate_md( md_alg );
+    unsigned char transcript[MBEDTLS_MD_MAX_SIZE];
```

Comment:
```suggestion
    unsigned char transcript[PSA_HASH_MAX_SIZE];
```
this takes also into account what drivers can do if any.